workflow/scripts/sbs_process/eval.py

```diff
@@ -0,0 +1,91 @@
+import pandas as pd
+import matplotlib.pyplot as plt
+
+from lib.sbs_process.eval import (
+    load_and_concatenate_hdfs,
+    plot_mapping_vs_threshold,
+    plot_mapping_vs_threshold,
+    plot_read_mapping_heatmap,
+    plot_cell_mapping_heatmap,
+    plot_reads_per_cell_histogram,
+    plot_gene_symbol_histogram,
+)
+
+# Read barcodes
+df_design = pd.read_csv(snakemake.params.df_design_path, sep="\t")
+df_pool = df_design.query("dialout==[0,1]").drop_duplicates("sgRNA")
+df_pool["prefix"] = df_pool.apply(lambda x: x.sgRNA[: x.prefix_length], axis=1)
+barcodes = df_pool["prefix"]
+
+# Concatenate files
+reads = load_and_concatenate_hdfs(snakemake.input.read_files)
+cells = load_and_concatenate_hdfs(snakemake.input.cell_files)
+minimal_phenotype_info = load_and_concatenate_hdfs(
+    snakemake.input.minimal_phenotype_info_files
+)
+
+plot_mapping_vs_threshold(reads, barcodes, "peak")
+plt.gcf().savefig(snakemake.output[0])
+plt.close()
+
+plot_mapping_vs_threshold(reads, barcodes, "Q_min")
+plt.gcf().savefig(snakemake.output[1])
+plt.close()
+
+plot_read_mapping_heatmap(reads, barcodes, shape="6W_sbs")
+plt.gcf().savefig(snakemake.output[2])
+plt.close()
+
+df_summary_one, _ = plot_cell_mapping_heatmap(
+    cells,
+    minimal_phenotype_info,
+    barcodes,
+    mapping_to="one",
+    mapping_strategy="gene_symbols",
+    shape="6W_sbs",
+    return_summary=True,
+)
+df_summary_one.to_csv(snakemake.output[3], index=False, sep="\t")
+plt.gcf().savefig(snakemake.output[4])
+plt.close()
+
+df_summary_any, _ = plot_cell_mapping_heatmap(
+    cells,
+    minimal_phenotype_info,
+    barcodes,
+    mapping_to="any",
+    mapping_strategy="gene_symbols",
+    shape="6W_sbs",
+    return_summary=True,
+)
+df_summary_any.to_csv(snakemake.output[5], index=False, sep="\t")
+plt.gcf().savefig(snakemake.output[6])
+plt.close()
+
+outliers = plot_reads_per_cell_histogram(cells, x_cutoff=20)
+plt.savefig(snakemake.output[7])
+plt.close()
+
+outliers = plot_gene_symbol_histogram(cells, x_cutoff=30)
+plt.gcf().savefig(snakemake.output[8])
```

Comment:
So it seems for snakemake compatibility, actually having outputted figs (by functions like plot_gene_symbol_histogram) makes a lot of sense, that way it can be saved to the path specified in the snakemake file? Is this accurate? We should keep this standardized throughout.

Reply:
Yep, outputted figs is the move. Good for consistency with other scripts because we always load and save files outside of lib functions.

Made all of these helper functions return figures in https://github.com/cheeseman-lab/brieflow/pull/10/commits/f460dd8242afd36458b3c8843873826652a760b2